delfin/drivers/dell_emc/vnx/vnx_block/component_handler.py

```diff
@@ -565,3 +572,340 @@ def list_storage_hosts(self, storage_id):
                 }
                 host_list.append(host_model)
         return host_list
+
+    def collect_perf_metrics(self, storage_id, resource_metrics,
+                             start_time, end_time):
+        metrics = []
+        archive_file_list = []
+        try:
+            if (end_time - start_time) < consts.EXEC_TIME_INTERVAL:
+                LOG.warn("not exe collert, end_time-start_time={}".format(
+                    (end_time - start_time)))
+                return metrics
+            archive_file_list = self._get__archive_file(start_time, end_time)
+            LOG.info("Get archive files: {}".format(archive_file_list))
+            if not archive_file_list:
+                LOG.warn("The required performance file was not found!")
+                return metrics
+            resources_map, resources_type_map = self._get_resources_map(
+                resource_metrics)
+            if not resources_map or not resources_type_map:
+                LOG.warn("Resource object not found!")
+                return metrics
+            performance_lines_map = self._filter_performance_data(
+                archive_file_list, resources_map, start_time, end_time)
+            if not performance_lines_map:
+                LOG.warn("The required performance data was not found!")
+                return metrics
+            metrics = self.create_metrics(storage_id, resource_metrics,
+                                          resources_map, resources_type_map,
+                                          performance_lines_map)
+        except exception.DelfinException as err:
+            err_msg = "Failed to collect metrics from VnxBlockStor: %s" % \
+                      (six.text_type(err))
+            LOG.error(err_msg)
+            raise err
+        except Exception as err:
+            err_msg = "Failed to collect metrics from VnxBlockStor: %s" % \
+                      (six.text_type(err))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+        finally:
+            self._remove_archive_file(archive_file_list)
+        return metrics
+
+    def create_metrics(self, storage_id, resource_metrics, resources_map,
+                       resources_type_map, performance_lines_map):
+        metrics = []
+        for resource_obj in resources_type_map.keys():
+            if not resources_map.get(resource_obj) \
+                    or not resources_type_map.get(resource_obj):
+                continue
+            resources_type = resources_type_map.get(resource_obj)
+            labels = {
+                'storage_id': storage_id,
+                'resource_type': resources_type,
+                'resource_id': resources_map.get(resource_obj),
+                'type': 'RAW',
+                'unit': ''
+            }
+            if not performance_lines_map.get(resource_obj):
```

Comment:
This `if` should be defined before the `labels` statement.

Reply:
Modified